src/utilities.jl

```diff
@@ -11,7 +11,12 @@ the proportion of nonzeros exceeds `threshold`.
 function densify(A::SparseMatrixCSC, threshold::Real = 0.25)
     m, n = size(A)
     if m == n && isdiag(A)  # convert diagonal sparse to Diagonal
-        Diagonal(diag(A))
+        if diag(A) isa SparseVector
+            # sparse diagonals don't make sense
+            Diagonal(Vector(diag(A)))
+        else
+            Diagonal(diag(A))
+        end
```

Comment:
Would it be better to write this chunk as
```julia
        Diagonal(Vector(diag(A)))
```
The call to `Vector` should be a no-op if `diag(A)` is already a dense vector.